core/commands/wallet.go

```diff
@@ -52,24 +57,81 @@ withdraw and query balance of token used in BTFS.`,
 		"transfer":          walletTransferCmd,
 		"discovery":         walletDiscoveryCmd,
 		"validate_password": walletCheckPasswordCmd,
+		"generate_key":      walletGenerateKeyCmd,
 	},
 }
 
 var walletInitCmd = &cmds.Command{
 	Helptext: cmds.HelpText{
-		Tagline:          "Init BTFS wallet",
-		ShortDescription: "Init BTFS wallet.",
+		Tagline:          "initialize BTFS wallet",
+		ShortDescription: "initialize BTFS wallet",
+	},
+	Arguments: []cmds.Argument{
+		cmds.StringArg("privateKey", true, false, "private key"),
+		cmds.StringArg("encrypted_private_key", true, false, "encrypted private key"),
+		cmds.StringArg("encrypted_mnemonic", true, false, "encrypted mnemonic"),
+	},
+	Options: []cmds.Option{},
+	Run: func(req *cmds.Request, res cmds.ResponseEmitter, env cmds.Environment) error {
+		n, err := cmdenv.GetNode(env)
+		if err != nil {
+			return err
+		}
+		cfg, err := n.Repo.Config()
+		if err != nil {
+			return err
+		}
+		cfg.Identity.PrivKey = req.Arguments[0]
+		cfg.Identity.Mnemonic = ""
+		cfg.Identity.EncryptedPrivKey = req.Arguments[1]
+		cfg.Identity.EncryptedMnemonic = req.Arguments[2]
+		ks, err := crypto.ToPrivKey(req.Arguments[0])
+		if err != nil {
+			return err
+		}
+		id, err := peer.IDFromPrivateKey(ks)
+		if err != nil {
+			return err
+		}
+		cfg.Identity.PeerID = id.String()
+		cfg.UI.Wallet.Initialized = true
+		if err = n.Repo.SetConfig(cfg); err != nil {
+			return err
+		}
+		go doRestart()
```

Comment:
in `doRestart` add a little bit of sleep, just in case the restart killed the return of this function.